src/vsphere_cpi/lib/cloud/vsphere/selection_pipeline.rb

```diff
@@ -0,0 +1,85 @@
+module VSphereCloud
+  class SelectionPipeline
+    include Enumerable
+
+    # @return [Object] the object to select placements for
+    attr_reader :object
+
+    # @param object [Object] the object to select placements for
+    def initialize(object, *args)
+      raise ArgumentError, 'No gather block provided' unless block_given?
```

Comment:
I want to be able to enforce design and developer constraints by raising an error so that any new pipeline developer needs to provide a block. 
What advantage does an empty list offer? We are not allowing for addition of gather block later.